src/ckeditor.jsx

```diff
@@ -5,24 +5,67 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import EditorWatchdog from '@ckeditor/ckeditor5-watchdog/src/editorwatchdog';
+import uid from '@ckeditor/ckeditor5-utils/src/uid';
+import { ContextWatchdogContext } from './context.jsx';
+import ContextWatchdog from '@ckeditor/ckeditor5-watchdog/src/contextwatchdog';
 
 export default class CKEditor extends React.Component {
 	constructor( props ) {
 		super( props );
 
 		// After mounting the editor, the variable will contain a reference to the created editor.
 		// @see: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editor-Editor.html
-		this.editor = null;
 		this.domContainer = React.createRef();
+
+		/**
+		 * An instance of EditorWatchdog if the context watchdog was not passed via React's context. `null` otherwise.
+		 */
+		this.watchdog = null;
+
+		/**
+		 * An instance of ContextWatchdog passed from the React's context or `null` otherwise.
+		 */
+		this.contextWatchdog = null;
```

Comment:
Looks like those variables keep the same value. The only difference between them is the source. The first one comes from the CKEditor component itself, the second one comes with CKEditor properties.

Couldn't we have a single (protected) property for a case when the component itself creates the context?

I'd add to a getter:

```js
get watchdog() {
	return this._watchdogContext || this.props.context; // I am unsure about names.
};
```

Reply:
I was thinking about a similar way. The proposed solution won't work as both solutions have slightly different API in case of getting the editor (as one watchdog is for the editor and the second is for the context.

Reply:
I've just written a small adapter for the Context Watchdog feature. I hope you'll like it.

Reply:
That's good.